package.json

```diff
@@ -45,5 +46,10 @@
     "tailwindcss": "^3.4.1",
     "ts-node": "^10.9.2",
     "typescript": "^5"
-  }
+  },
+  "pre-commit": [
+    "lint",
+    "prettier:format",
+    "test"
```

Comment:
I believe we can ignore `test`, as tests increase this would hinder dev experience.
We'll always have PR level checks for this.

Reply:
ideally , one would always run tests before committing to ensure his/her changes doesn't break anything.

Reply:
What I meant is we do develop with shorter commits in local and squash before commiting to PR/remote branch. Hence wanted to avoid this. Similar pattern is followed in `dice` repo for pre-commits.

Reply:
sure I'll do the necessary changes